service/storage_proxy.cc

```diff
@@ -3408,8 +3400,15 @@ storage_proxy::create_write_response_handler_helper(schema_ptr s, const dht::tok
         db::consistency_level cl, db::write_type type, tracing::trace_state_ptr tr_state, service_permit permit, db::allow_per_partition_rate_limit allow_limit, is_cancellable cancellable, coordinator_mutate_options options) {
     replica::table& table = _db.local().find_column_family(s->id());
     auto erm = table.get_effective_replication_map();
-    host_id_vector_replica_set natural_endpoints = erm->get_natural_replicas(token);
-    host_id_vector_topology_change pending_endpoints = erm->get_pending_replicas(token);
+
+    host_id_vector_replica_set natural_endpoints;
+    host_id_vector_topology_change pending_endpoints;
+    if (options.node_local_only) {
+        natural_endpoints = host_id_vector_replica_set{my_host_id(*erm)};
```

Comment:
What if it's not part of the original natural_endpoints? We might access a node that does not have a tablet replica, which violates an invariant.

Reply:
commit message:
> If the current node is not a replica, writes will hit apply_on_shards and fail with "No local shards for token". For reads, sharder.shard_for_reads currently returns shard 0 by default (there is already a TODO item to remove this default), and the read will fail later when no storage group is found for the tablet on shard 0.

Reply:
OK, I'll do the checks here.